src/main/java/com/tabnine/general/StaticConfig.java

```diff
@@ -127,9 +137,9 @@ public static String getTabNineVersionUrl() {
   }
 
   @NotNull
-  public static String getTabNineBundleVersionUrl() {
+  public static String getTabNineBundleVersionUrl() throws NoValidBinaryToRunException {
     return Optional.ofNullable(System.getProperty(REMOTE_VERSION_URL_PROPERTY))
-        .orElse(getBundleServerUrl() + "/version");
+        .orElse(getBundleServerUrl().orElseThrow(NoValidBinaryToRunException::new) + "/version");
```

Comment:
im not sure about the error handling here - what happens if `NoValidBinaryToRunException` is thrown here? does the user see the ugly popup with this exception?
i think you should instead `warn` log this case, and return `Optional<String>` here, and the user should see the gray (inactive) status bar if the URL is not set (this is already implemented)